lib/db.js

```diff
@@ -453,6 +481,78 @@ module.exports = function (
     })
   }
 
+  DB.prototype.createDevice = function (uid, sessionTokenId, deviceInfo) {
+    log.trace({ op: 'DB.createDevice', uid: uid, id: deviceInfo.id })
+    var self = this
+    deviceInfo.id = crypto.randomBytes(16)
+    deviceInfo.createdAt = Date.now()
+    return this.pool.put(
+      '/account/' + uid.toString('hex') +
+      '/device/' + deviceInfo.id.toString('hex'),
+      unbuffer({
+        sessionTokenId: sessionTokenId,
+        createdAt: deviceInfo.createdAt,
+        name: deviceInfo.name,
+        type: deviceInfo.type,
+        callbackURL: deviceInfo.pushCallback,
+        callbackPublicKey: deviceInfo.pushPublicKey
+      })
+    )
+    .then(
+      function () {
+        return deviceInfo
+      },
+      function (err) {
+        if (isRecordAlreadyExistsError(err)) {
```

Comment:
nit: this could use a comment to explain why we're making another call to the db to check on things here